### ISSUE TITLE
Font type issues when performing global searches

### ISSUE THREAD

**Issue by IgnacioHeredia** [Zettlr/Zettlr#4896]
### Description

Hi,

I just updated to 3.0.3 and I'm experiencing some weird issues.

Sometimes, when I perform a global search and I click in the results, the resulting file opens and the whole text is in a weird monospace (large) font:

![Screenshot from 2024-01-13 11-58-26](https://github.com/Zettlr/Zettlr/assets/16413205/94e2858b-6010-47b3-8597-7ff16af03fa9)

But if I open the file from the File Explorer, font is rendered fine:

![Screenshot from 2024-01-13 11-59-12](https://github.com/Zettlr/Zettlr/assets/16413205/c5f84e9c-851d-461a-82aa-ef69df63d6ec)

I thought this might be related with the refactoring of themes. I use Bielefeld and I do use some minor CSS customizations, but noting related to font type. I copy the CSS below just in case.

One additional important thing is that, once you start _editing_ a badly rendered file, the font type returns to normal. 

Thanks!

<details>
<summary>Custom CSS</summary>

```css

/*
Use paired with Bielefeld theme in dark mode 
See also: https://github.com/Zettlr/Zettlr/discussions/2863
*/

/* ----------------- */
/* Light mode styles */
/* ----------------- */

.cm-editor .cm-quote {
  color: var(--c-primary);
}

.cm-editor .cm-emphasis {
  color: var(--c-primary);
}

/* ---------------- */
/* Dark mode styles */
/* ---------------- */

/* Background color */
body.dark .cm-editor {
   background-color: rgba(32, 33, 37, 1);
}

/* Generic tags */
body.dark .cm-editor .cm-zkn-tag {
   background-color: rgba(200, 200, 200, 1);
   color: rgba(74, 74, 74, 1) !important;
   padding: 2px;
   border-radius:5px;
}

/* Custom colors for custom tags */

.cm-zkn-tag-todo {
   background-color: rgba(275,171,171, 1);
   color: rgba(138,0,0, 1) !important;
}

.cm-zkn-tag-software {
   background-color: #ffe599;
   color:#7e6100 !important;
}

.cm-zkn-tag-sota, .cm-zkn-tag-done {
   background-color: #d8ead2 !important;
   color:#274e13 !important;
}

.cm-zkn-tag-cv, .cm-zkn-tag-nlp, .cm-zkn-tag-tabular, .cm-zkn-tag-aka, .cm-zkn-tag-tldr {
   background-color: #d8d3e9 !important;
   color:#21134d !important;
}

.cm-zkn-tag-easy, .cm-zkn-tag-medium, .cm-zkn-tag-hard, .cm-zkn-tag-resources  {
   background-color: #c8daf9 !important;
   color:#1c4587 !important;
}

/* Quotes */
body.dark .cm-editor .cm-quote {
   color: #deaa02;
}

/* Bold */
body.dark .cm-editor .cm-strong {
   color: rgba(182, 249, 250, 1);
}

/* Italics */
body.dark .cm-editor .cm-emphasis {
   color: rgba(255, 165, 230, 1);
}

/* Monospace */
body.dark .cm-editor .cm-monospace {
   color: rgba(255, 255, 255, 1);
   background-color: rgba(70, 70, 70, 1);
   padding: 2px;
   padding-right: 5px;
   padding-left: 5px;
   border-radius: 5px;
}

/*
Links 
 .cma - link preview 
 .cm-label-name - link text
 .cm-url - url link
*/
body.dark .cm-editor :is(.markdown-link, .cm-label-name, .cm-url) {
   color: rgba(0,204,153, 1) !important;
}

/* Bullet points */
body.dark .cm-editor .cm-list.cm-code-mark {
   color: rgba(255, 242, 0, 0.8);
}

/* Bold marks */
body.dark .cm-editor .cm-strong.cm-code-mark {
   color: rgba(204, 204, 204, 0.4);
}

/* Italic marks */
body.dark .cm-editor .cm-emphasis.cm-code-mark {
   color: rgba(204, 204, 204, 0.4);
}

/* Horizontal rule */
body.dark .cm-editor .cm-hr {
    display: inline-block;
    width: 100%;
    line-height: 0.25;
    color: transparent;
    background: linear-gradient(90deg, var(--c-primary),  #7beac3, transparent);
}

/* Internal links */
body.dark .cm-editor .cm-zkn-link:not(:hover) {
	font-size: .01em;
  	visibility: hidden;
    color: rgba(0,204,153, 1);
}
body.dark .cm-editor .cm-zkn-link:not(:hover)::after {
    font-size: 100em;
    content: '↗';
    visibility: visible;
    color: rgba(0,204,153, 1);
}
body.dark .cm-editor .cm-zkn-link {
    color: rgba(0,204,153, 1) !important;
}

/* IDs */
body.dark .cm-editor .cm-zkn-id:not(:hover) {
	font-size: .01em;
  	visibility: hidden;
    color: rgba(252, 186, 3, 1);
}
body.dark .cm-editor .cm-zkn-id:not(:hover)::after {
    font-size: 100em;
    /*content: '\1F194';*/
    content: '\1F3F7 ID';
    visibility: visible;
    font-weight: bold;
    color: rgba(252, 186, 3, 1);
}
body.dark .cm-editor .CodeMirror .cm-zkn-id {
    color: rgba(252, 186, 3, 1);
}
```

</details>

### Reproducing

No sure how to reproduce this as it happens in searches _from time to time_ (let's say 50% of the time), not consistently. I haven't found a pattern yet. Will report here if I'm able to reproduce consistently. In the _same search_, some results can be rendered correctly and other be rendered incorrectly.

### Zettlr Version

Stable (most recent version)

### Specify version

3.0.3

### Installation Method

From the Website or GitHub

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Linux 20.04

### Additional Information

Installed via deb package.

**Comment by khakhalin** [comment]
For me (3.0.3 on Windows) the font doesn't return to normal once I start editing (remains large monospace). Also, the cursor isn't shown, so while I can edit, I do it kinda "blindly". For now the only workaround is to remember the file name, and then find it anew by name, through the file panel, but of course it's somewhat suboptimal :)

**Comment by IgnacioHeredia** [comment]
Indeed. I ended up downgrading back to 3.0.2 because this was a bit annoying. Hopefully this will get fixed in new releases. 🤞🏽

**Comment by nathanlesage** [comment]
This likely has nothing to do with the search per se, sometimes there are issues with loading the editor, and if something fails, the theme doesn't load properly, and that explains the weird monospace font (that's the basic styles without any theme applied).

Please try out the coming nightly on Monday and see if that fixes the issue. I couldn't reproduce right now.

**Comment by khakhalin** [comment]
Yep, confirming that it's not search-related; search just seems to make it more likely that it happens. Loading longer documents with lots of formatting also seems to be a factor.

**Comment by nathanlesage** [comment]
Quick update: maybe don't try out the Monday nightly just yet, I am currently sifting through the entirety of the application GUI and while doing that I uncovered a ton of bugs; I'll try to fix them but I probably won't finish today.

**Comment by IgnacioHeredia** [comment]
This seems to be solved in 3.1.1 (at least for me).
